# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteVideoDecoder.h
# Commit: 557f55a90c8a
# Full Hash: 557f55a90c8aa215b7046fadf7a17436baa93713
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-06-27 09:34:48
# Regressor Bug: 1561178
# File Overlap Count: 1
# Description:
#   Bug 1561178 - Make passing a TextureFactoryIdentifier optional, and pass one whenever we can. r=jya
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35965
# ==============================================================================

diff -r e1521d8fbe90 -r 557f55a90c8a dom/media/ipc/RemoteVideoDecoder.h
--- a/dom/media/ipc/RemoteVideoDecoder.h	Thu Jun 27 01:28:47 2019 +0000
+++ b/dom/media/ipc/RemoteVideoDecoder.h	Thu Jun 27 01:28:56 2019 +0000
@@ -25,7 +25,8 @@
 
   MOZ_IS_CLASS_INIT
   MediaResult InitIPDL(const VideoInfo& aVideoInfo, float aFramerate,
-                       const CreateDecoderParams::OptionSet& aOptions);
+                       const CreateDecoderParams::OptionSet& aOptions,
+                       const layers::TextureFactoryIdentifier* aIdentifier);
 
   IPCResult RecvOutput(const DecodedOutputIPDL& aDecodedData) override;
 
@@ -54,13 +55,12 @@
 
 class RemoteVideoDecoderParent final : public RemoteDecoderParent {
  public:
-  RemoteVideoDecoderParent(RemoteDecoderManagerParent* aParent,
-                           const VideoInfo& aVideoInfo, float aFramerate,
-                           const CreateDecoderParams::OptionSet& aOptions,
-                           const layers::TextureFactoryIdentifier& aIdentifier,
-                           TaskQueue* aManagerTaskQueue,
-                           TaskQueue* aDecodeTaskQueue, bool* aSuccess,
-                           nsCString* aErrorDescription);
+  RemoteVideoDecoderParent(
+      RemoteDecoderManagerParent* aParent, const VideoInfo& aVideoInfo,
+      float aFramerate, const CreateDecoderParams::OptionSet& aOptions,
+      const Maybe<layers::TextureFactoryIdentifier>& aIdentifier,
+      TaskQueue* aManagerTaskQueue, TaskQueue* aDecodeTaskQueue, bool* aSuccess,
+      nsCString* aErrorDescription);
 
  protected:
   MediaResult ProcessDecodedData(
